lib/Sema/CSDiagnostics.h

```diff
@@ -2306,6 +2306,28 @@ class NonEphemeralConversionFailure final : public ArgumentMismatchFailure {
   void emitSuggestionNotes() const;
 };
 
+class SendingOnFunctionParameterMismatchFail final : public ContextualFailure {
+public:
+  SendingOnFunctionParameterMismatchFail(const Solution &solution, Type srcType,
+                                         Type dstType,
+                                         ConstraintLocator *locator,
+                                         FixBehavior fixBehavior)
+      : ContextualFailure(solution, srcType, dstType, locator, fixBehavior) {}
+
+  bool diagnoseAsError() override;
+};
+
+class SendingOnFunctionResultMismatchFailure final : public ContextualFailure {
```

Comment:
It looks like this split is not necessary, the diagnostic is anchored on a simplified locator so it would point to the right spot.